### DISCUSSION TITLE
Vehicle Max deceleration

### DISCUSSION THREAD

**Discussion by fabio-r-goncalves** [eclipse/mosaic#336] [General]
Hi,

I am running a simulation with multiple vehicle types with different accelerations and decelerations configured in the Mapping file. However, it seems that the vehicles do not use the maximum acceleration I have defined in the mapping file. Despite the vehicle type or my configuration, they all seem to have 4.5 maximum deceleration. On the other hand, the configurations seem to work for the maximum acceleration. An example vehicle:

 { 
                    "name": "Motorcycle3",
                    "vehicleClass": "Motorcycle",
                    "length": 2.50,
                    "accel": 4.0,
                    "decel": 9.0,
                    "maxSpeed": 30.0,
                    "color": "#00cc99",
                    "applications": ["com.uminho.infrarob.applications.SimpleVehicleCommsApp"],
                    "weight": 0.5
}

Thanks in advance.

**Comment by schwepmo** [comment]
I just did a quick test in the Tiergarten-scenario and set `"accel"` and `"decel"` to 9.0 and at least from checking in the sumo-gui it seems as though the parameters are properly applied.

![grafik](https://github.com/eclipse/mosaic/assets/39306374/747a5458-936c-48d5-be82-1c2ea825f6cc)

How are you measuring/checking the deceleration in your scenario?

OR in case you are using an existing SUMO-scenario you cannot set these parameters via the mapping file but instead have to adjust the `vType`-parameters which may be located within the route file or an additional file.

**Reply by fabio-r-goncalves** [reply]
Hi there,

I am using a scenario created by myself. All vehicles are introduced using matrixMappers.
In your case, it seems ok, but I have just tested it, and it all defaults to 4.5. The same happens to the maximum speed. Do I have something ill-defined?

![image](https://github.com/eclipse/mosaic/assets/5545859/6d7e72c3-84e3-40c9-bc3b-0637cec56fb9)

Here is the mapping corresponding to that specific vehicle type:

![image](https://github.com/eclipse/mosaic/assets/5545859/213b1ef0-8bea-4254-b4fe-b8720c199c41)

I am using Mosaic version 22.1 with SUMO 1.15.0

**Reply by kschrab** [reply]
1. In your screenshots the type ids do not match, please check this again.
2. For `decel` I can confirm that there is an error. For `maxSpeed` it looks fine however.
3. The problem you describe happens _only_ when you specify your type in the `types` section of the spawner/matrixMapper. However, if you define all your types in the main `prototypes` section, and then refer to these types in your specific `types` sections, it should (hopefully) work:

```
{
"prototypes": [
  { name: "MyCar1", decel: 9.0, applications: [...], ... },
  { name: "MyCar2", decel: 8.5, applications: [...], ... }
],
"matrixMappers": [
  { types: [ { name: "MyCar1", weight: 0.1 }, { name: "MyCar2", weight: 0.9 } ], 
  ...
  }
 ]
}
```




You

**Reply by kschrab** [reply]
@schwepmo I created a [PR](https://github.com/eclipse/mosaic/pull/337) which should fix this problem

**Reply by fabio-r-goncalves** [reply]
Moving the vehicle specification solved the problem. Thank you so much.
